src/classes/App.php

```diff
@@ -79,7 +79,11 @@ public function __construct(Request $request, Session $session, Config $config,
     {
         $this->Request = $request;
         $this->Session = $session;
-        $this->ok = $this->Session->getFlashBag()->get('ok');
+
+        // Don't get 'ok' flashes during mfa
```

Comment:
why is this needed? wouldn't it be better to make sure nothing is sent to the ok bag during mfa? (it's always better to avoid ifs)

Reply:
A user needs to enable 2FA in the _Account_ tab of the _User Control Panel_. At the same time other settings can be changed. Before this PR, this changes are confirmed with a 'Saved' message. However, with 2FA enabled there will be a few redirects (UcpController.php -> login.php -> LoginController.php -> ucp.php?tab=2). A User will get the secret and can setup their app, eLab will verify the 2FA code and enable 2FA. On the way many things can go wrong but the other setting are stored first. Therefore, the 'Saved' message is added to the flashbag directly (open to discussion).
I see several options here:
1. Show the 'Saved' message on the login.php page. **Pro**: no need to do the if. **Con**: Maybe it is confusing for the user to get the message here.
2. Don't add the 'Saved' message at all if user also wants to enable 2FA. **Pro**: no need to do the if. **Con**: User gets no feedback that the other settings are stored. But can see it later in the UCP.
3. Only add the 'Saved' message after 2FA was successfully enabled. **Pro**: no need to do the if. **Con**: User only gets feedback if all works as intended but no feedback in any other case.
4. Maybe it is possible to use a Session variable? I would need to think about it a bit. **Pro**: no need to do the if. **Con**: ?

Reply:
I'll need to run your code and see by myself how it can be done.